### PR TITLE
Divide IN_POOL tx state into EXECUTED and PENDING states

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Divide IN_POOL tx state into EXECUTED and PENDING states

Related issues: https://github.com/ainblockchain/ain-blockchain/issues/484

### CODE REVIEW

File: tx-pool/index.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ class TransactionPool {
     this.txCountTotal = 0;
   }
 
-  addTransaction(tx) {
+  addTransaction(tx, isExecutedTx = false) {
```
Comment (liayoo) [inline tx-pool/index.js:42]: question) I noticed that we aren't actually setting isExecutedTx = true anywhere we call addTransaction() yet. Is this intended?

Comment (platfowner) [reply tx-pool/index.js:42]: Right. Some critical part was mistakenly missing. Just fixed and added some test cases. Thanks.

**Comment by platfowner** [comment]
Comments addressed. PTAL @liayoo

**Review by liayoo — APPROVED** [review]
LGTM, thanks!

**Pull request merged**